connector/rest/src/main/java/it/pagopa/selfcare/external_api/connector/rest/MsCoreConnectorImpl.java

```diff
@@ -120,4 +125,23 @@ public List<UserProducts> getOnboarderUsers(List<TokenUser> users) {
         log.trace("getOnboarderUsers end");
         return onboardedUsers.stream().map(userProductMapper::toEntity).collect(Collectors.toList());
     }
+
+    @Override
+    public List<OnboardedInstitutionInfo> getInstitutionDetails(String institutionId) {
+        ResponseEntity<InstitutionResponse> responseEntity = institutionApiClient._retrieveInstitutionByIdUsingGET(institutionId);
+        if (Objects.nonNull(responseEntity)) {
+            InstitutionResponse response = responseEntity.getBody();
+            if (Objects.nonNull(response) && Objects.nonNull(response.getOnboarding())) {
+                return response.getOnboarding().stream().map(onboardedProductResponse -> {
+                    OnboardedInstitutionInfo onboardedInstitutionInfo = institutionMapper.toOnboardedInstitution(response);
+                    ProductInfo productInfo = institutionMapper.toProductInfo(onboardedProductResponse);
+                    onboardedInstitutionInfo.setProductInfo(productInfo);
+                    onboardedInstitutionInfo.setState(productInfo.getStatus());
+                    return onboardedInstitutionInfo;
+                }).toList();
+            }
+        }
+        return Collections.emptyList();
```

Comment:
you could execute only one if, for ex:

`if (Objects.isNull(responseEntity) || Objects.isNull(responseEntity.getBody()) || Objects.isNull(responseEntity.getBody().getOnboarding())  return Collections.emptyList()`